### PR TITLE
fix(main): pass env to spawn #1295

### LINKED ISSUE #1295: bug: Cannot load init.lua using NVIM_APPNAME with WSL
### Did you check docs and existing issues?

- [X] I have read all the vscode-neovim docs
- [X] I have searched the existing issues of vscode-neovim
- [X] I can reproduce the issue with an empty init.vim/init.lua and no other vscode extensions (when applicable)

### Neovim version (nvim -v)

0.9.1

### Operating system/version

Windows 11 (WSL2)

### Describe the bug

When using NVIM_APPNAME, vscode-neovim does not load init.lua via NVIM_APPNAME.

### Steps To Reproduce

1. Delete ~/.config/nvim/init.lua.
2. Set NVIM_APPNAME to `my-app`.
3. Create ~/.config/my-app/init.lua:
   ```lua
   vim.keymap.set('n', '<S-l>', 'gt', { remap = true })
   ```
4. Start VSCode, press <kbd>Shift</kbd> + <kbd>l</kbd>, nothing happens.

### Expected Behavior

Trigger the `gt` keybinding.

### PR DESCRIPTION
**Pull request by theol0403**
Fixes #1295

### CODE REVIEW

**Comment by jackblackevo** [comment]
I found that when `settings.useWsl` is set to true, the `child_process.spawn()` function is actually as follows:

```javascript
child_process.spawn(
  'C:\\Windows\\system32\\wsl.exe',
  [
    '$HOME/.local/bin/nvim',
    '-N',
    '--embed',
    '-c',
    'source /mnt/c/Users/jackblackevo/.vscode/extensions/asvetliakov.vscode-neovim-0.3.1/runtime/lua/options.lua',
    '--cmd',
    'source /mnt/c/Users/jackblackevo/.vscode/extensions/asvetliakov.vscode-neovim-0.3.1/vim/vscode-neovim.vim'
  ],
  { env: process.env }
);
```

Due to this configuration, the environment variables cannot be passed to Neovim because Neovim is running in WSL.

However, I tried using Windows `cmd.exe` to simulate the behavior of `child_process.spawn()` and passed `NVIM_APPNAME=my-app` to Neovim, and it worked:

```shell
wsl.exe NVIM_APPNAME=my-app $HOME/.local/bin/nvim
```

I can successfully use the Vim command `:echo $MYVIMRC` to display the correct path: `/home/my-username/.config/my-app/init.lua`.

**Comment by justinmk** [comment]
`--cmd` happens early enough that using `:let` works to set NVIM_APPNAME. So this should work:

```
'--cmd',
`let $NVIM_APPNAME='${settings.NVIM_APPNAME}'`,
```

**Comment by jackblackevo** [comment]
Thank you, @justinmk!

First, I tried the following command in Windows `cmd.exe`:

```shell
wsl.exe $HOME/.local/bin/nvim --cmd "let $NVIM_APPNAME='my-app'"
```

However, I received an error message from Neovim:

```text
Error detected while processing pre-vimrc command line:
E475: Invalid argument: ='my-app'
Press ENTER or type command to continue
```

So, I added a backslash (`\`) and tried again:

```shell
wsl.exe $HOME/.local/bin/nvim --cmd "let \$NVIM_APPNAME='my-app'"
```

I modified the source code and added the following code, then built it locally:

```javascript
if (settings.NVIM_APPNAME) {
  args.push("--cmd", `let \\$NVIM_APPNAME='${settings.NVIM_APPNAME}'`);
}
```

It worked successfully!

**Comment by justinmk** [comment]
Nice, thanks for confirming. But the `\$` is only needed in bash, not nodejs.

**Comment by jackblackevo** [comment]
@justinmk I tried removing `\$`, but it was unable to load `init.lua`.

```javascript
args.push("--cmd", `let NVIM_APPNAME='${settings.NVIM_APPNAME}'`);
```

**Comment by jackblackevo** [comment]
Update:

Based on the following code snippet:

```javascript
if (settings.NVIM_APPNAME) {
  args.push("--cmd", `let \\$NVIM_APPNAME='${settings.NVIM_APPNAME}'`);
}
```

I added additional Neovim configuration files in `~/.config/my-app/`, and then attempted to require the 'plugins' module:

```text
/lua/plugins.lua
/init.lua
```

However, I encountered an error in the VSCode Output panel:

```text
Error detected while processing /home/my-username/.config/my-app/init.lua:
E5113: Error while calling lua chunk: /home/my-username/.config/my-app/init.lua:1: module 'plugins' not found:
  no field package.preload['plugins']
  no file './plugins.lua'
  no file '/__w/neovim/neovim/.deps/usr/share/luajit-2.1.0-beta3/plugins.lua'
  no file '/usr/local/share/lua/5.1/plugins.lua'
  no file '/usr/local/share/lua/5.1/plugins/init.lua'
  no file '/__w/neovim/neovim/.deps/usr/share/lua/5.1/plugins.lua'
  no file '/__w/neovim/neovim/.deps/usr/share/lua/5.1/plugins/init.lua'
  no file './plugins.so'
  no file '/usr/local/lib/lua/5.1/plugins.so'
  no file '/__w/neovim/neovim/.deps/usr/lib/lua/5.1/plugins.so'
  no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
  [C]: in function 'require'
  /home/my-username/.config/my-app/init.lua:1: in main chunk
```

**Comment by justinmk** [comment]
> @justinmk I tried removing `\$`, but it was unable to load `init.lua`.

The "$" is needed. Not the `\`.

**Comment by jackblackevo** [comment]
> > @justinmk I tried removing `\$`, but it was unable to load `init.lua`.
> 
> The "$" is needed. Not the `\` (unless javascript flags it as a syntax error?)
> 
> ```
> args.push("--cmd", `let $NVIM_APPNAME='${settings.NVIM_APPNAME}'`);
> ```

It is causing an error in the VSCode Output panel:

```text
Error detected while process pre-vimrc command line:
E475: Invalid argument: ='my-app'
```

**Comment by jackblackevo** [comment]
I tried using the shell variable `NVIM_APPNAME=my-app` in the following code:

```typescript
child_process.spawn(
  'C:\\Windows\\system32\\wsl.exe',
  [
    'NVIM_APPNAME=my-app',
    '$HOME/.local/bin/nvim',
    '-N',
    '--embed',
    '-c',
    'source /mnt/c/Users/jackblackevo/.vscode/extensions/asvetliakov.vscode-neovim-0.3.1/runtime/lua/options.lua',
    '--cmd',
    'source /mnt/c/Users/jackblackevo/.vscode/extensions/asvetliakov.vscode-neovim-0.3.1/vim/vscode-neovim.vim'
  ],
  {}
);
```

```diff
 if (settings.useWsl) {
   args.unshift(settings.neovimPath);
 }
+if (settings.NVIM_APPNAME) {
+  if (settings.useWsl) {
+    args.unshift(`NVIM_APPNAME=${settings.NVIM_APPNAME}`);
+  } else {
+    process.env.NVIM_APPNAME = settings.NVIM_APPNAME;
+  }
+}
```

This approach enabled me to correctly require the 'plugins' module.

However, I am unsure why `--cmd let $NVIM_APPNAME=my-app` doesn't work.

**Comment by theol0403** [comment]
Possibly fixed now?

File: src/main_controller.ts [hunk 1]
```diff
@@ -139,10 +146,9 @@ export class MainController implements vscode.Disposable {
                 settings.useWsl
             }, args: ${JSON.stringify(args)}`,
         );
```
Comment (jackblackevo) [inline src/main_controller.ts:148]: Maybe we can log the `env` for debugging convenience.

**Pull request closed without merging**